### PR TITLE
readme: split DEVELOPMENT.md; add Sphinx site

### PR DESCRIPTION
**Pull request by umarcor**
The README contains several outdated references to Travis CI. This PR splits contributing/building guidelines to file DEVELOPMENT.md, and updates the content. CI workflows are updated accordingly.

### CODE REVIEW

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -0,0 +1,170 @@
+# Building
```
Comment (proppy) [inline DEVELOPMENT.md:1]: nit: Development? (to match the filename)

**Review by proppy — APPROVED** [review]
just a nit

File: DEVELOPMENT.md (hunk 1 above)
Comment (umarcor) [reply DEVELOPMENT.md]: Done.

File: docs/DEVELOPMENT.md [hunk 2]
```diff
@@ -0,0 +1,170 @@
+# Development
+
+This repository is set up to be built in GitHub Actions.
+See [`.github/workflows/build-packages.yml`](.github/workflows/build-packages.yml).
+
+On a successful build, the resulting packages are uploaded to https://anaconda.org/litex-hub/repo and can be installed
+with: `conda install --channel "LiteX-Hub" package`.
+
+These packages are mostly used by [`litex-buildenv`](https://github.com/litex-hub/litex-buildenv).
```
Comment (PiotrZierhoffer) [inline docs/DEVELOPMENT.md:9]: I think this is not true anymore - they are being used in tons of places which are more up to date, so I'd drop the reference.

Comment (umarcor) [reply docs/DEVELOPMENT.md:9]: Actually, I did not update the content (yet). I just moved it and put it in a site. Let's update it in a follow-up PR, along with adding refs to other conda-* repos.

**Pull request merged**